test/utils/padder/padder.go

```diff
@@ -0,0 +1,212 @@
+	namespace string
+	*padRequest
+}
+
+type padRequest struct {
+	nNodes           int
+	allocationTarget corev1.ResourceList
+}
+
+// New return new padder object
+// ns is where padding pods will get deployed
+func New(cli client.Client, ns string) (*Padder, error) {
+	return &Padder{
+		Client:     cli,
+		namespace:  ns,
+		padRequest: &padRequest{allocationTarget: make(corev1.ResourceList)},
+	}, nil
+}
+
+// Nodes number of nodes to pad
+func (p *Padder) Nodes(n int) *Padder {
+	p.padRequest.nNodes = n
+	return p
+}
+
+// UntilAvailableIsResource will pad pods into nodes until reach the desired available allocationTarget
+func (p *Padder) UntilAvailableIsResource(resName corev1.ResourceName, quantity string) *Padder {
+	quan := resource.MustParse(quantity)
+	p.allocationTarget[resName] = quan
+	return p
+}
+
+// UntilAvailableIsResourceList is like UntilAvailableIsResource but with a complete ResourceList as a parameter
+func (p *Padder) UntilAvailableIsResourceList(resources corev1.ResourceList) *Padder {
+	p.allocationTarget = resources
+	return p
+}
+
+// Pad will create pad pods in order to align the nodes
+// with the requested amount of available allocationTarget
+func (p *Padder) Pad() error {
+	if p.nNodes == 0 {
+		klog.Warningf("no nodes for padding were found. please specify at least one node")
+		return nil
+	}
+
+	nrtList, err := nrtutil.GetUpdated(p.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Second*10)
+	if err != nil {
+		return err
+	}
+
+	singleNumaNrt := filterSingleNumaNodePolicyNrts(nrtList.Items)
+
+	for i := 0; i < len(singleNumaNrt) && p.nNodes > 0; i++ {
+		nrt := singleNumaNrt[i]
+		nodePadded := false
+		for _, zone := range nrt.Zones {
+			// check that zone has at least the amount of allocationTarget that needed
+			if nrtutil.ZoneResourcesMatchesRequest(zone.Resources, p.allocationTarget) {
+				availResList := nrtutil.AvailableFromZone(zone)
+				diffList, err := diffAvailableToExpected(availResList, p.allocationTarget)
+				if err != nil {
+					return err
+				}
+
+				// create a pod that asks for exact amount of allocationTarget as the diff
+				// in order to reach to desired amount of available allocationTarget in the node
+				padPod := objects.NewTestPodPause(p.namespace, fixture.RandomName("padder"))
+
+				// label the pod with a pad label, so it will be easier to identify later
+				labelPod(padPod, map[string]string{PadderLabel: ""})
+
+				padPod.Spec.Containers[0].Resources.Limits = diffList
+				padPod.Spec.Containers[0].Resources.Requests = diffList
+
+				// place the pod on the selected node
+				//padPod.Spec.NodeSelector = map[string]string{nodes.LabelHostname: nrt.Name}
+				padPod.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nrt.Name}
+				if err := p.Client.Create(context.TODO(), padPod); err != nil {
+					return err
+				}
+				if _, err = wait.ForPodPhase(p.Client, p.namespace, padPod.Name, corev1.PodRunning, time.Minute); err != nil {
```

Comment:
uhm, do we want to wait sequentially for each pod to go running (or fail)?

Reply:
we can make that in parallel indeed. let me rewrite that part

Reply:
@fromanirh on second thought i'm not sure that having this function inside a go routine is a good idea.
How do we report an error? using a channel is overkill in my opinion. 
having panic or assertion outside of ginkgo spec is also a bad idea. WDYT?